4_naming_conventions/43_file_names.md

```diff
@@ -0,0 +1,212 @@
+```
+<Feature><Phase>.inf
+
+   <Feature>      REQUIRED    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm, StandaloneMm,
+                              Smm, Uefi.
+
+Example:
+   - SmbiosDxe.inf
+```
+
+* If the implementation of CPU arch or vendor shares some code in the module. Or the implementation is only for the specific CPU arch or vendor:
+```
+<Feature><Phase><CpuArch><Vendor>.inf
+
+   <Feature>      REQUIRED    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm,
+                              StandaloneMm, Smm, Uefi.
+   <CpuArch>      OPTIONAL    The <CpuArch> is represented with a BNF,
+                              <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                        'LoongArch64' | 'Ebc'
+                              <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                              Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>       OPTIONAL    *
+
+Example:
+   - CpuIo2DxeAmd.inf
+   - CpuIo2DxeIntel.inf
+   - CpuIo2DxeIa32_X64.inf
+
+```
+
+* If the implementation does not have any shared code between phases:
+
+```
+<Feature>/<Phase>/<Feature>.inf
+  or
+<Feature>/<Phase>/<Feature><CpuArch><Vendor>.inf
+
+   <Feature>      OPTIONAL    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm, StandaloneMm, Smm,
+                              Uefi.
+   <CpuArch>      OPTIONAL    The <CpuArch> is represented with a BNF,
+                              <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                        'LoongArch64' | 'Ebc'   
+                              <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                              Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>       OPTIONAL    *
+Example:
+   PCD/Dxe/Pcd.inf
+```
+
+#### 4.4.5.3 EDK II INF file within a Library instance
+```
+<Phase><CpuArch><Vendor><LibraryClassName><Dependency>.inf
+   <Phase>              REQUIRED     Base, Sec, Pei, Dxe, DxeRuntime, Mm,
+                                     StandaloneMm, Smm, Uefi.
+   <CpuArch>            OPTIONAL     The <CpuArch> is represented with a BNF,
+                                     <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                               'LoongArch64' | 'Ebc'
+                                     <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                                     Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>             OPTIONAL     *
+   <LibraryClassName>   REQUIRED     *
+   <Dependency>         OPTIONAL     * (Typically name of PPI, Protocol, LibraryClass
+                                       that the implementation is layered on top of)
+
+Example:
+   - SmmAmdSmmCpuFeaturesLib.inf
+   - SmmIntelSmmCpuFeaturesLib.inf

```

Comment:
I still think this should be SmmAmdCpuFeaturesLib.inf. Since we can't change existing SmmCpuFeaturesLib.inf and CpuCommonFeaturesLib.inf, they can be considered as Intel related if different from AMD.

Reply:
What we define here is generic and mainly for the new module that is introduced to edk2
From the naming guidelines perspective, SmmAmdSmmCpuFeatureLib is compliant with <phase><CpuArch><Vendor><LibraryClassName><Dependency> file naming structure. So both SmmAmdSmmCpuFeaturesLib and SmmIntelSmmCpuFeaturesLib are correct.
However, SmmCpuFeaturesLib was implemented before the guidelines, the way to reconstruct the existing module is mentioned here, [The Guidelines of Reconstruct EDK II Modules for Processor Architectures and Vendors' Implementation](https://github.com/changab/tianocore.github.io/wiki/The-Guidelines-of-Reconstruct-EDK-II-Modules-for-Processor-Architectures-and-Vendors'-Implementation).
Take SmmCpuFeaturesLib as an example, we won't touch the existing INF file. However, the newly added INF file will follow the guidelines. I will remove SmmIntelSmmCpuFeaturesLib as I know that is confusing, but I will keep AMD one because that is a new file.

Reply:
Okay.